jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonb/JsonbDiagnosticsCollector.java

```diff
@@ -82,17 +82,20 @@ private void collectJsonbTransientFieldDiagnostics(ICompilationUnit unit,
             List<Diagnostic> diagnostics) throws JavaModelException {
         for (IType type : unit.getAllTypes()) {
             for (IField field : type.getFields()) {
-                boolean hasJsonbTransient = false, hasOtherJsonbAnnotation = false;
-                for (IAnnotation annotation : field.getAnnotations()) {
-                    String annotation_name = annotation.getElementName();
-                    if (annotation_name.equals(JsonbConstants.JSONB_TRANSIENT))
-                        hasJsonbTransient = true;
-                    else if (annotation_name.startsWith(JsonbConstants.JSONB_PREFIX))
-                        hasOtherJsonbAnnotation = true;
-                }
-                if (hasJsonbTransient && hasOtherJsonbAnnotation) {
-                    Diagnostic diagnostic = createDiagnosticBy(unit, field, JsonbConstants.JSONB_TRANSIENT);
-                    diagnostics.add(diagnostic);
+                if (hasJsonbTransient(field)) {
+                    // Check if the field itself is annotated with other Jsonb annotations.
+                    if (hasNonJsonbTransientAnnotation(field)) {
+                        Diagnostic diagnostic = createDiagnosticBy(unit, field, JsonbConstants.JSONB_TRANSIENT);
+                        diagnostics.add(diagnostic);
+                    }
+                    // Diagnostics on the getter and setter of the field are created when they are
+                    // annotated with Jsonb annotations other than JsonbTransient.
+                    for (IMethod accessor : JDTUtils.getAccessors(field)) {
+                        if (hasNonJsonbTransientAnnotation(accessor)) {
+                            Diagnostic diagnostic = createDiagnosticBy(unit, accessor, JsonbConstants.JSONB_TRANSIENT);
```

Comment:
Can we make the diagnostic message clearer on what the problem is/how developers could resolve the issue. 
The current diagnostic appears as: 
```
JsonbTransient must be mutually exclusive with all other JSON Binding defined annotations.
```

but can we adjust the wording to call out the getter/setter? Similar to how the initial issue describes:
"If a class field is annotated with @JsonbTransient, exception is thrown when this field, getter or setter is annotated with other JSON Binding annotations."

Reply:
I adjusted the wording.